test/test_helpers.py

```diff
@@ -39,9 +39,9 @@ def cleanup():
     with (test_case.fake_script_path / name).open('wt') as f:
         # Before executing the provided script, dump the provided arguments in calls.txt.
         # ASCII 1E is RS 'record separator', and 1C is FS 'file separator', which seem appropriate.
-        f.write('''#!/bin/bash
-{ echo -n $(basename $0); printf "\\x1e%s" "$@"; printf "\\x1c"; } >> $(dirname $0)/calls.txt
-''' + content)
+        f.write('''#!/bin/sh
+{{ printf {}; printf "\\036%s" "$@"; printf "\\034"; }} >> {}/calls.txt
```

Comment:
I had looked at shell, and noted that \x1e wasn't supported by plain sh. Interesting that 036 works in its place.
If you're going to template it, though, I'd prefer the **kwargs style, since it is clearer to read what bits go into what section:
```
template_args = {
  'name': name,
  'path': path,
  'content': content,
}
f.write('''#!/bin/sh
{{ printf {name}; printf "\\036a%s" "$@"; printf "\\034"; }} >> {path}/calls.txt
{content}```.format(**template_args)
```

Thoughts?

Reply:
agreed it's more readable; done.

Reply:
actually went with `format_map` instead :-)